### PR TITLE
Add supplier class

### PR DESCRIPTION
**Pull request by qlchan24**

### CODE REVIEW

File: docs/tutorials/AddRemark.md [hunk 1]
```diff
@@ -263,21 +263,21 @@ That’s it! Fire up the application again and you should see something like thi
...
+Since `SupplierCard` displays data from a `Supplier`, we need to update `Supplier` to get our `Remark` displayed!
 
-### Modify `Person`
+### Modify `Supplier`
 
-We change the constructor of `Person` to take a `Remark`. We will also need to define new fields and accessors accordingly to store our new addition.
+We change the constructor of `Supplier` to take a `Remark`. We will also need to define new fields and accessors accordingly to store our new addition.
 
-### Update other usages of `Person`
+### Update other usages of `supplier`
```
Comment (jeffreytjs) [inline docs/tutorials/AddRemark.md:274]: Minor issue, remember to capitalize Supplier and line 276 and 280 as well.

File: docs/tutorials/TracingCode.md [hunk 2]
```diff
@@ -180,15 +180,15 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
...
+           source.getSupplierList()
                  .stream()
-                 .map(JsonAdaptedPerson::new)
+                 .map(JsonAdaptedSupplier::new)
                  .collect(Collectors.toList()));
    }
    ```
 
-1. It appears that a `JsonAdaptedPerson` is created for each `Person` and then added to the `JsonSerializableAddressBook`.
+1. It appears that a `JsonAdaptedSupplier` is created for each `supplier` and then added to the `JsonSerializableAddressBook`.
```
Comment (jeffreytjs) [inline docs/tutorials/TracingCode.md:191]: Same issue, minor typo for Supplier

File: docs/tutorials/TracingCode.md [hunk 3]
```diff
@@ -245,6 +245,6 @@ the given commands to find exactly what happens.
 
     4.  Add a new command
 
-    5.  Add a new field to `Person`
+    5.  Add a new field to `supplier`
```
Comment (jeffreytjs) [inline docs/tutorials/TracingCode.md:248]: Capitalize also

File: src/main/java/seedu/address/model/supplier/UniqueSupplierList.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+     * The supplier must exist in the list.
+     */
+    public void remove(Supplier toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new SupplierNotFoundException();
+        }
+    }
+
+    public void setSuppliers(UniqueSupplierList replacement) {
```
Comment (jeffreytjs) [inline src/main/java/seedu/address/model/supplier/UniqueSupplierList.java:82]: Similar to comments left on UniqueWarehouseList, should this be removed instead?

File: src/main/java/seedu/address/model/supplier/UniqueSupplierList.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the supplier {@code target} in the list with {@code editedSupplier}.
+     * {@code target} must exist in the list.
+     * The supplier identity of {@code editedSupplier} must not be the same as another existing supplier in the list.
+     */
+    public void setSupplier(Supplier target, Supplier editedSupplier) {
```
Comment (jeffreytjs) [inline src/main/java/seedu/address/model/supplier/UniqueSupplierList.java:56]: Similar to comments left on UniqueWarehouseList, might want to change to replaceSupplier

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 6]
```diff
@@ -11,24 +11,24 @@
...
+import seedu.address.model.supplier.Supplier;
 
 /**
  * An Immutable AddressBook that is serializable to JSON format.
  */
 @JsonRootName(value = "addressbook")
 class JsonSerializableAddressBook {
 
-    public static final String MESSAGE_DUPLICATE_PERSON = "Persons list contains duplicate person(s).";
+    public static final String MESSAGE_DUPLICATE_SUPPLIER = "suppliers list contains duplicate supplier(s).";
```
Comment (jeffreytjs) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:22]: Minor issue, capitalise Suppliers

File: src/main/java/seedu/address/ui/SupplierCard.java [hunk 7]
```diff
@@ -1,20 +1,18 @@
...
 
 /**
- * An UI component that displays information of a {@code Person}.
+ * An UI component that displays information of a {@code Supplier}.
  */
-public class PersonCard extends UiPart<Region> {
+public class SupplierCard extends UiPart<Region> {
 
-    private static final String FXML = "PersonListCard.fxml";
+    private static final String FXML = "supplierListCard.fxml";
```
Comment (jeffreytjs) [inline src/main/java/seedu/address/ui/SupplierCard.java:15]: Capitalise Suppliers here

**Review by tohyuting — APPROVED** [review]
LGTM

**Pull request merged**